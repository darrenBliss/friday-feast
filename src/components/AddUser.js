// @flow
import * as React from 'react'
import styled from 'styled-components'
import { Card } from '../style/Card'
import { Button } from '../style/Button'

const AddHolder = styled.form`
  display: flex;
  input {
    flex: 1 1 auto;
    margin-right: .2rem;
  }
`

type Props = {
  onAddUser: (string) => void
}

type State = {
  value: string
}

class AddUser extends React.Component<Props, State> {
  addUser: () => void
  handleChange: () => void

  constructor() {
    super()

    this.state = {
      value: ''
    }

    this.addUser = this.addUser.bind(this)
    this.handleChange = this.handleChange.bind(this)
  }

  addUser(event: SyntheticEvent<>) {
    event.preventDefault()

    const trimmedVal: string = this.state.value.trim()

    if (trimmedVal) {
      this.props.onAddUser(trimmedVal)
      this.setState({
        value: ''
      })
    }
  }

  handleChange(event: SyntheticInputEvent<HTMLInputElement>) {
    this.setState({
      value: event.currentTarget.value
    })
  }

  render() {
    return (
      <Card>
        <AddHolder onSubmit={this.addUser}>
          <input name="user-name"
            value={this.state.value}
            onChange={this.handleChange}
            type="text" />
          <Button type="submit">Add person</Button>
        </AddHolder>
      </Card>
    )
  }
}

export default AddUser