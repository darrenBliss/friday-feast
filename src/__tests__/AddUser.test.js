import React from 'react'
import { shallow, mount } from 'enzyme'
import toJson from 'enzyme-to-json'

import AddUser from '../components/AddUser'

describe('AddUser component', () => {
  it('Render AddUser snapshot', () => {
    const wrapper = shallow(<AddUser />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('Should have input field', () => {
    const wrapper = shallow(<AddUser />)

    expect(wrapper.find('input')).toHaveLength(1)
  })

  it('State change should work', () => {
    const wrapper = mount(<AddUser />)

    const theInput = wrapper.find('input')
    theInput.instance().value = 'Megatron'
    theInput.simulate('change')

    expect(wrapper.state('value')).toEqual('Megatron')
  })

  it('Form submit should call with value')
  it('Form submit should reset state value')





})
