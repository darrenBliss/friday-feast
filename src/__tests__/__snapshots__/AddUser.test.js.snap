// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddUser component Render AddUser snapshot 1`] = `
<Card>
  <AddUser__AddHolder
    onSubmit={[Function]}
  >
    <input
      name="user-name"
      onChange={[Function]}
      type="text"
      value=""
    />
    <Button
      type="submit"
    >
      Add person
    </Button>
  </AddUser__AddHolder>
</Card>
`;
